src-tauri/src/tray/updater.rs

```diff
@@ -12,29 +12,39 @@ pub fn check_for_update(app_handle: AppHandle) -> Result<()> {
                     Ok(update_option) => {
                         if let Some(update) = update_option {
                             let update_str = format!(
-                                "Current Version: {}\nLatest Version: {}",
+                                "Your Version: {}\nLatest Version: {}",
                                 update.current_version, update.version
                             );
 
-                            println!("{}", update_str);
+                            println!("update available:\n\tdownload url: {}", update.download_url);
 
                             app_handle
                                 .dialog()
                                 .message(update_str)
-                                .title("A new version of Helmer Micro is out!")
+                                .title("A new version is available!")
                                 .ok_button_label("Install")
                                 .cancel_button_label("Cancel")
-                                .show(|result| match result {
-                                    true => {
-                                        // TODO: install update
-                                        // update.download_and_install(None, None);
-                                    }
-                                    _ => {}
+                                .show(move |result| {
+                                    tauri::async_runtime::spawn(async move {
+                                        match result {
+                                            true => {
+                                                let _ = update
+                                                    .clone()
+                                                    .download_and_install(
```

Comment:
I don't think this method here works very well. Will need to figure out if there are better ways of approaching the install, ideally without much frontend work.